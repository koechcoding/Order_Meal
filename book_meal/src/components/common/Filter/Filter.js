import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import './styles.css';

class Filter extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            text: '',
        };
        this.inputRef = React.createRef();
    }

    onReset = (e) => {
        this.setState({
            ...this.state,
            text: '',
        });
        this.inputRef.current.value = '';
        this.props.onFilter('')
    }

    onChange = (e) => {
        this.setState({
            ...this.state,
            text: e.target.value
        });
        this.props.onFilter(e.target.value)
    }

    render() {
        return (
            <div className="row mb-3">
                <div className="col-3 col-sm-2 col-lg-1 text-right">
                    <label className="pt-2">Text:</label>
                </div>
                <div className="col-9 col-sm-7 col-lg-4 col-xl-3 pl-0 pr-1">
                    <input 
                        ref={this.inputRef} 
                        onChange={_.throttle(this.onChange, 500)} 
                        type="text" 
                        className="d-inline form-control ml-0 mb-0"  placeholder="Type to filter..."/>
                </div>
                {this.state.text &&
                    <div className="animated fadeIn btn-reset col-12 col-sm-1 col-lg-1 pt-3 pt-sm-0 pl-0">
                        <button onClick={this.onReset} className="btn btn-secondary d-block m-auto">Reset </button>
                    </div>
                }
            </div>
        );
    }
};

Filter.propTypes = {
    onFilter: PropTypes.func.isRequired,
};

export default Filter;