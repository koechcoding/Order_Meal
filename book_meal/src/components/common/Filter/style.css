
.btn-reset.animated {
    animation-duration: .3s; }