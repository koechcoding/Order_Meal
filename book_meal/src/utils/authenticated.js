export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}