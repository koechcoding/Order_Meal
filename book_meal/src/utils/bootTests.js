import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
