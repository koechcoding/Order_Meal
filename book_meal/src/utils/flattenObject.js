export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}