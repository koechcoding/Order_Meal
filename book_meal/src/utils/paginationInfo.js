export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}