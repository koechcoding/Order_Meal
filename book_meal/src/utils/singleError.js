const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;