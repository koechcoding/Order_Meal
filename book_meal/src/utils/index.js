import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}