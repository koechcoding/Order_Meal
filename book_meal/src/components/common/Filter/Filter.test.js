import React from 'react';
import { shallow, mount} from 'enzyme';
import 'src/utils/bootTests';
import Filter from './filter';
import Adapter from 'enzyme-adapter-react-16';

describe('<Filter/>', ()=>{
    it('renders correctly', ()=>{
        const wrapper = shallow(<Filter onFilter={()=>{}}/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('accepts input', ()=>{
        const wrapper = shallow(<Filter onFilter={()=>{}}/>);
        wrapper.find('input').simulate('change', {target: {value: 'hi there'}});
        expect(wrapper.state('text')).toEqual('hi there');
    });
    it('accepts resetting', ()=>{
        const wrapper = mount(<Filter onFilter={()=>{}}/>);
        wrapper.find('input').simulate('change', {target: {value: "hi"}});
        wrapper.find('button').simulate('click');
        expect(wrapper.state('text')).toEqual('');
    });
});