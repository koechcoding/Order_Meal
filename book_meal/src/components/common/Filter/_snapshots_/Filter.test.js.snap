// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    onFilter={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-3 col-sm-2 col-lg-1 text-right"
>
          <label
                    className="pt-2"
          >
                    Text:
          </label>
</div>,
        <div
          className="col-9 col-sm-7 col-lg-4 col-xl-3 pl-0 pr-1"
>
          <input
                    className="d-inline form-control ml-0 mb-0"
                    onChange={[Function]}
                    placeholder="Type to filter..."
                    type="text"
          />
</div>,
        "",
      ],
      "className": "row mb-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label
            className="pt-2"
>
            Text:
</label>,
          "className": "col-3 col-sm-2 col-lg-1 text-right",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Text:",
            "className": "pt-2",
          },
          "ref": null,
          "rendered": "Text:",
          "type": "label",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="d-inline form-control ml-0 mb-0"
            onChange={[Function]}
            placeholder="Type to filter..."
            type="text"
/>,
          "className": "col-9 col-sm-7 col-lg-4 col-xl-3 pl-0 pr-1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "d-inline form-control ml-0 mb-0",
            "onChange": [Function],
            "placeholder": "Type to filter...",
            "type": "text",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-3 col-sm-2 col-lg-1 text-right"
>
            <label
                        className="pt-2"
            >
                        Text:
            </label>
</div>,
          <div
            className="col-9 col-sm-7 col-lg-4 col-xl-3 pl-0 pr-1"
>
            <input
                        className="d-inline form-control ml-0 mb-0"
                        onChange={[Function]}
                        placeholder="Type to filter..."
                        type="text"
            />
</div>,
          "",
        ],
        "className": "row mb-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label
              className="pt-2"
>
              Text:
</label>,
            "className": "col-3 col-sm-2 col-lg-1 text-right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Text:",
              "className": "pt-2",
            },
            "ref": null,
            "rendered": "Text:",
            "type": "label",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="d-inline form-control ml-0 mb-0"
              onChange={[Function]}
              placeholder="Type to filter..."
              type="text"
/>,
            "className": "col-9 col-sm-7 col-lg-4 col-xl-3 pl-0 pr-1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "d-inline form-control ml-0 mb-0",
              "onChange": [Function],
              "placeholder": "Type to filter...",
              "type": "text",
            },
            "ref": Object {
              "current": null,
            },
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;